staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -284,10 +285,28 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	return &authenticator.Response{
 		User: &user.DefaultInfo{
 			Name:   chain[0].Subject.CommonName,
+			UID:    parseUIDFromCert(chain[0]),
 			Groups: chain[0].Subject.Organization,
 			Extra: map[string][]string{
 				user.CredentialIDKey: {id},
 			},
 		},
 	}, true, nil
 })
+
+// Using CNCF PEN for Kubernetes:
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/4317-pod-certificates#x509-podidentity-extension
+var uidOID = asn1.ObjectIdentifier([]int{1, 3, 6, 1, 4, 1, 57683, 2})
+
+// Go's crypto/x509 library doesn't explicitly serialize UID to a named field, so
+// we read it off the asn1 array.
+func parseUIDFromCert(cert *x509.Certificate) string {
```

Comment:
same for this

Reply:
Addressed in this discussion: https://github.com/kubernetes/kubernetes/pull/127897#discussion_r1825031398